modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -257,14 +257,22 @@ func (p *ProjectPipelineService) Create(ctx context.Context, params *pb.CreatePr
 		PipelineSourceID: sourceRsp.PipelineSource.ID,
 		Category:         DefaultCategory.String(),
 		Extra: &dpb.PipelineDefinitionExtra{
-			Extra: p.pipelineSourceType.GetPipelineCreateRequestV2(),
+			Extra: pipelineSourceType.GetPipelineCreateRequestV2(),
 		},
 		Ref: sourceRsp.PipelineSource.Ref,
 	})
 	if err != nil {
 		return nil, apierrors.ErrCreateProjectPipeline.InternalError(err)
 	}
 
+	// When creating a definition, a scheduled task is created asynchronously
+	go func() {
+		err := p.createCronIfNotExist(definitionRsp.PipelineDefinition, pipelineSourceType)
```

Comment:
any compensate mechanism if create failed?

Reply:
There is no open transaction, even if it is not asynchronous, there is an inconsistency problem

Reply:
what will happen if cron create failed? The invoker **DO NOT** know whether the cron success or not, and you return success even when cron failed.
It will cause cron task cannot be executed as expected, and it's serious enough.

Reply:
I changed to synchronous call